source/gui/__init__.py

```diff
@@ -483,49 +484,90 @@ def onInstallCommand(self, evt):
+				return
+			else:
+				log.error("Could not execute fixCOMRegistrations command", exc_info=True)
+				error = e  # Hold for later display to the user
+				return  # Safe because of finally block
+		except Exception:
 			log.error("Could not execute fixCOMRegistrations command", exc_info=True)
-		progressDialog.done()
-		del progressDialog
+			return
+		finally:  # Clean up the progress dialog, and display any important error to the user before returning
+			progressDialog.done()
+			del progressDialog
+			self.postPopup()
+			# If there was a Windows error, inform the user because it may have support value
+			if error is not None:
+				messageBox(
+					_(
+						# Translators: message shown to the user on COM Registration Fix fail
+						"The COM Registration Fixing Tool was unsuccessful. This Windows "
+						"error may provide more information.    {}",
+					).format(error),
+					# Translators: Added to the title of the dialog showing the COM Registration Fix failure
+					genericTitle + " " + _("Failed"),
+					wx.OK,
+				)
+		# Display success dialog if there were no errors
 		messageBox(
 			_(
-				# Translators: The message displayed when the COM Registration Fixing tool completes.
-				"The COM Registration Fixing tool has finished. "
+				# Translators: Message shown when the COM Registration Fixing tool completes.
+				"The COM Registration Fixing Tool has completed successfully.\n"
 				"It is highly recommended that you restart your computer now, to make sure the changes take full effect.",
 			),
-			# Translators: The title of a dialog presented when the COM Registration Fixing tool is complete.
-			_("COM Registration Fixing Tool"),
+			genericTitle,
 			wx.OK,
 		)
 
+	def _CRFTIntroMessage(self) -> str:
+		"""Helper that returns the COM Registration Fixing Tool's introductory statement."""
+		return _(
+			# Translators: Explain the COM Registration Fixing tool to users before running
+			"Welcome to the COM Registration Fixing tool.\n"
+			"This tool is used by NVDA to fix problems it may have as it tries to interact with various applications, "
+			"or with Windows itself.\n"
+			"It examines the system registry for corrupted or missing accessibility entries and will correct them.\n"
+			"Those entries are sometimes damaged by installing or uninstalling programs, or other system events. This "
+			'can result in "unknown" or "pane" being spoken instead of the content you were expecting, or previously '
+			"accessible elements suddenly no longer reading correctly.\n\n"
+			"You have most likely been asked to run this tool by an experienced user or support worker to investigate and potentially fix an issue with NVDA's behaviour.\n"
+			"Because it needs to modify the Windows registry, if you have User Account Control (UAC) active, you will "
+			"be prompted by UAC before this tool can do its job. This is normal and you should answer by pressing the "
+			"yes button.\n\n"
```

Comment:
Users whose account doesn't have administrative privileges will need to enter the admin password. These instructions should either be modified to account for this, or removed entirely.